src/domain/community/profile/profile.resolver.mutations.ts

```diff
@@ -1,79 +1,71 @@
-import { ProfileInput } from './profile.dto';
 import { ProfileService } from './profile.service';
-import { IProfile } from '@domain/community/profile/profile.interface';
-import { Profile } from '@domain/community/profile';
 
 @Resolver()
-export class ProfileResolver {
+export class ProfileResolverMutations {
   constructor(private profileService: ProfileService) {}
 
   @Roles(AuthorizationRoles.CommunityAdmins, AuthorizationRoles.EcoverseAdmins)
   @UseGuards(GqlAuthGuard)
   @Mutation(() => Tagset, {
-    description:
-      'Creates a new tagset with the specified name for the profile with given id',
+    description: 'Creates a new Tagset on the specified Profile',
   })
   @Profiling.api
   async createTagsetOnProfile(
-    @Args('profileID') profileID: number,
-    @Args('tagsetName') tagsetName: string
+    @Args('tagsetData') tagsetData: CreateTagsetInput
   ): Promise<ITagset> {
-    const tagset = await this.profileService.createTagset(
-      profileID,
-      tagsetName
-    );
-    return tagset;
+    return await this.profileService.createTagset(tagsetData);
   }
 
   @Roles(AuthorizationRoles.CommunityAdmins, AuthorizationRoles.EcoverseAdmins)
   @UseGuards(GqlAuthGuard)
   @Mutation(() => Reference, {
-    description:
-      'Creates a new reference with the specified name for the profile with given id',
+    description: 'Creates a new Reference on the specified Profile.',
   })
   @Profiling.api
   async createReferenceOnProfile(
-    @Args('profileID') profileID: number,
-    @Args('referenceInput') referenceInput: ReferenceInput
+    @Args('referenceInput') referenceInput: CreateReferenceInput
   ): Promise<IReference> {
-    const reference = await this.profileService.createReference(
-      profileID,
-      referenceInput
-    );
-    return reference;
+    return await this.profileService.createReference(referenceInput);
   }
 
   @Roles(AuthorizationRoles.EcoverseAdmins, AuthorizationRoles.CommunityAdmins)
   @UseGuards(GqlAuthGuard)
-  @Mutation(() => Boolean, {
-    description:
-      'Updates the fields on the Profile, such as avatar location or description',
+  @Mutation(() => Profile, {
+    description: 'Updates the specified Profile.',
   })
   @Profiling.api
   async updateProfile(
-    @Args('ID') profileID: number,
-    @Args('profileData') profileData: ProfileInput
-  ): Promise<boolean> {
-    return await this.profileService.updateProfile(profileID, profileData);
+    @Args('profileData') profileData: UpdateProfileInput
+  ): Promise<IProfile> {
+    return await this.profileService.updateProfile(profileData);
   }
 
   @Roles(AuthorizationRoles.EcoverseAdmins, AuthorizationRoles.CommunityAdmins)
   @SelfManagement()
-  @Mutation(() => Profile)
+  @Mutation(() => Profile, {
+    description: 'Uploads and sets an avatar image for the specified Profile.',
+  })
   async uploadAvatar(
-    @Args('profileID') profileID: number,
+    @Args('uploadData') uploadData: UploadProfileAvatarInput,
```

Comment:
I think I misled you here - we can put the GraphQLUpload inside the UploadProfileAvatarInput

Reply:
Had tried but was having an issue - can you give it a go?